CorsixTH/Lua/graphics.lua

```diff
@@ -590,90 +590,117 @@ function AnimationManager:setAnimLength(anim, length)
-      frame = anims:getNextFrame(frame)
+      frame = self.anims:getNextFrame(frame)
     end
     self.anim_length_cache[anim] = length
   end
   return self.anim_length_cache[anim]
 end
 
 --[[ Markers can be set using a variety of different arguments:
-  setMarker(anim_number, position)
-  setMarker(anim_number, start_position, end_position)
-  setMarker(anim_number, keyframe_1, keyframe_1_position, keyframe_2, ...)
-
-  position should be a table; {x, y} for a tile position, {x, y, "px"} for a
-  pixel position, with (0, 0) being the origin in both cases.
-
-  The first variant of setMarker sets the same marker for each frame.
-  The second variant does linear interpolation of the two positions between
-  the first frame and the last frame.
-  The third variant does linear interpolation between keyframes, and then the
-  final position for frames after the last keyframe. The keyframe arguments
-  should be 0-based integers, as in the animation viewer.
+  set{Patient,Staff}Marker(anim_number, position)
+  set{Patient,Staff}Marker(anim_number, {position1, position2, ..., positionN})
+  set{Patient,Staff}Marker(anim_number, start_position, end_position)
+  set{Patient,Staff}Marker(anim_number, keyframe_1, keyframe_1_position, keyframe_2, ...)
+
+  The 'position' should be a table; An {x, y} pair for a tile position or
+  {x, y, "px"} for a pixel position. In both cases the position should be the
+  humanoid centre at floor level, with the centre of the (0, 0) tile being the
+  origin in both cases.
+
+  The first variant of set{Patient,Staff}Marker sets the same marker for each frame.
+  The second variant of has a unique position for each frame. 'nil' can be used to
+  repeat the previous position.
+  The third variant does linear interpolation of the two positions between the first
+  frame and the last frame.
+  The fourth variant does linear interpolation between keyframes, and then the final
+  position for frames after the last keyframe. The keyframe arguments should be
+  0-based integers, as in the animation viewer.
 
   To set the markers for multiple animations at once, the anim_number argument
   can be a table, in which case the marker is set for all values in the table.
   Alternatively, the values function (defined in utility.lua) can be used in
   conjection with a for loop to set markers for multiple things.
 --]]
 
-function AnimationManager:setMarker(anim, ...)
-  return self:setMarkerRaw(anim, "setFramePrimaryMarker", ...)
+--! Define the centre of a patient at floor level in an animation.
+function AnimationManager:setPatientMarker(anim, ...)
+  self:_unfoldAnims(anim, "setFramePrimaryMarker", ...)
+end
+
+--! Define the centre of a staff member at floor level in an animation.
+function AnimationManager:setStaffMarker(anim, ...)
+  self:_unfoldAnims(anim, "setFrameSecondaryMarker", ...)
 end
 
-local function TableToPixels(t)
-  if t[3] == "px" then
-    return t[1], t[2]
+local function positionToXy(pos)
```

Comment:
So many coordinate systems- can you document this.

Reply:
Actually it's already explained somewhat higher in the file near the explanation of the types of marker settings that you can do.

I documented this function.

Reply:
We lack a @see tag I think, otherwise that would be appropriate. Read the other explanation and it's quite helpful. This is good.